sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuBroadcastExchangeExec.scala

```diff
@@ -145,32 +154,38 @@ class SerializeConcatHostBuffersDeserializeBatch(
    * NOTE: The caller is responsible to release these host columnar batches.
    */
   def hostBatches: Array[ColumnarBatch] = this.synchronized {
-    batchInternal match {
-      case batch if batch == null =>
-        withResource(new NvtxRange("broadcast manifest batch", NvtxColor.PURPLE)) { _ =>
-          val columnBatches = new mutable.ArrayBuffer[ColumnarBatch]()
-          closeOnExcept(columnBatches) { cBatches =>
-            headers.zip(buffers).foreach { case (header, buffer) =>
-              val hostColumns = SerializedHostTableUtils.buildHostColumns(
-                header, buffer, dataTypes)
-              val rowCount = header.getNumRows
-              cBatches += new ColumnarBatch(hostColumns.toArray, rowCount)
-            }
+    Option(batchInternal).map { maybeSpillable =>
+      val cols = maybeSpillable.fold(
+        GpuColumnVector.extractColumns,
+        sp => withResource(sp.getColumnarBatch())(GpuColumnVector.extractColumns)
+      )
+      val hostColumns = cols.map(_.copyToHost())
```

Comment:
Shouldn't we use safeMap here to copy the data to host so that the things are closed if there is an exception.

Reply:
sounds good. This is from the original code https://github.com/NVIDIA/spark-rapids/pull/6604/files#diff-7ca82be5615724f3ba4376b82be1c8a34e3fb5df34a3dedfa9a21eb3e6e54b97L163